src/tasks/minions/alchingActivity.ts

```diff
@@ -0,0 +1,46 @@
+import { AlchingActivityTaskOptions } from '../../lib/types/minions';
+import { Task } from 'klasa';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import { channelIsSendable } from '../../lib/util/channelIsSendable';
+import { addBanks, resolveNameBank, toKMB } from 'oldschooljs/dist/util';
+
+export default class extends Task {
+	async run({
+		itemName,
+		quantity,
+		channelID,
+		alchValue,
+		userID,
+		savedRunes
+	}: AlchingActivityTaskOptions) {
+		const user = await this.client.users.fetch(userID);
+
+		await user.settings.update(UserSettings.GP, user.settings.get(UserSettings.GP) + alchValue);
+
+		// If there are any saved runes from bryophyta's staff, return them
+		if (savedRunes) {
+			const returnedRunes = resolveNameBank({
+				'Nature rune': savedRunes
+			});
+			const userBank = await user.settings.get(UserSettings.Bank);
+			const newBank = await addBanks([userBank, returnedRunes]);
+			await user.settings.update(UserSettings.Bank, newBank);
+		}
+
+		const channel = this.client.channels.get(channelID);
+		if (!channelIsSendable(channel)) return;
+
+		const responses = [
+			`${user}, ${
+				user.minionName
+			} has finished alching ${quantity}x ${itemName}! ${alchValue.toLocaleString()}gp (${toKMB(
+				alchValue
+			)}) has been added to your bank`,
+			savedRunes > 0 ? `Your Bryophyta's staff saved you ${savedRunes} Nature runes` : ''
+		];
+
+		this.client.queuePromise(() => {
+			channel.send(responses.join('. '));
+		});
+	}
```

Comment:
traditionally any activity that takes time and can cause someone to not be able to do all of what they want to in one trip, the option to resend the trip that was just completed by responding with "yes"/"y" is something they are given.  take a look at other activities for examples